src/middlewares/authMiddleware.js

```
@@ -0,0 +1,22 @@
+const { jwtService } = require('../services/jwt.service');
+const { ApiError } = require('../exeptions/api.error.js');
+
+const authMiddleware = (req, res, next) => {
+  const authorization = req.headers['authorization'] || '';
+  const [, token] = authorization.split(' ');
+
+  if (!authorization || !token) {
+    return next(ApiError.unauthorized());
+  }
+
+  const userData = jwtService.verify(token);
```

`jwtService.verify(token)` returns `null` for invalid/expired tokens (the jwt service swallows verify exceptions and returns null). If you want to accept a valid refresh token when the access token is expired, you should attempt `verifyRefresh` (or reuse the repository's `authOnly` middleware which already implements that fallback). Otherwise users with expired access tokens but valid refresh tokens will be rejected. See `jwt.service.js` and `authOnly.js` for reference.